contrib/glpk-cut-log.patch

```diff
@@ -589,23 +593,32 @@ index 926dda4..55adf44 100644
 +           xerror("glp_ios_pool_size: operation not allowed\n");
 +      }
        xassert(tree->local != NULL);
-       return tree->local->size;
- }
-diff --git a/src/glpios.h b/src/glpios.h
-index 9e2d6b2..3b31901 100644
---- a/src/glpios.h
-+++ b/src/glpios.h
-@@ -36,6 +36,9 @@ typedef struct IOSAIJ IOSAIJ;
- typedef struct IOSPOOL IOSPOOL;
+ #ifdef NEW_LOCAL /* 02/II-2018 */
+       return tree->local->m;
+diff --git a/src/draft/ios.h b/src/draft/ios.h
+index e2b47c6..2e61fdc 100644
+--- a/src/draft/ios.h
++++ b/src/draft/ios.h
+@@ -24,7 +24,7 @@
+ 
+ #include "prob.h"
+ 
+-#if 1 /* 02/II-2018 */
++#if 0 /* 02/II-2018 */
+ #define NEW_LOCAL 1
+ #endif
```

Comment:
Ditto

Reply:
In GLPK 4.65, GLPK had this change:

> * src/draft/ios.h, src/draft/glpios01.c, src/draft/glpios03.c 
> These modules were changed to use the new cover cut generator.
>

[https://fossies.org/linux/glpk/ChangeLog]

This "new cover cut generator" isn't compatible with Tim's changes, but it is behind a `#define` to allow you get the old behaviour.

This `#if 0` changes GLPK's `#if 1` such that GLPK 5.0 uses the _old_ cut generator to maintain compatibility with Tim's patches.

I added this comment against the `#if 0`:

```
cvc5; NEW_LOCAL is not compatible with cut logging
```

Should I be more explicit or is that enough?

It might be worthwhile seeing if I can get Tim's stuff working with `NEW_LOCAL`, but I think that is future work.

Reply:
@mpreiner I'm leaving this one unresolved so you can check you're happy with the explanation.

Reply:
That's fine, thanks for the explanation!